src/coverage-system/sectionfinder.ts

```diff
@@ -101,53 +93,54 @@ export class SectionFinder {
-     * based on currently open workspaceFolders
-     * @param path potential partial path to be converted
+     * Calculates relativePath and workspaceFolder for given TextEditor instance
+     * Returned relativePath is normalized relative path (against workspace folder) of editor filename, starts with ###
+     * Returned workspaceFolder is name of workspaceFolder
+     * Ex. workspace: "c:/Workspace/testProject"
+     *     editor file: "c:/workspace/testProject/src/com/MyCompany/lib/Bla.java"
+     *     returned relativePath: "###src###com###mycompany###lib###bla.java"
+     *     returned workspaceFolder: "testProject"
+     * @param textEditor Instance of TextEditor
      */
-    private convertPartialPathsToAbsolute(path: string): string {
-        const files: string[] = [];
-        const globFind = (folder: string) => {
-            // find the path in the workspace folder
-            const possibleFiles = glob.sync(
-                `**/${path}`,
-                {
-                    cwd: folder,
-                    dot: true,
-                    ignore: this.configStore.ignoredPathGlobs,
-                    realpath: true,
-                },
-            );
-            if (possibleFiles.length) {
-                files.push(...possibleFiles);
-            }
-        };
-
-        if (!workspace.workspaceFolders) { return path; }
-        // Path is already absolute
-        // Note 1: some coverage generators can start with no slash #160
-        // Note 2: accounts for windows and linux style file paths
-        // windows as they start with drives (ie: c:\)
-        // linux as they start with forward slashes
-        // both windows and linux use ./ or .\ for relative
-        const unixRoot = path.startsWith("/");
-        const windowsRoot = path[1] === ":" && path[2] === "\\";
-        if (unixRoot || windowsRoot) {
-            return path;
-        }
-
-        // look over all workspaces for the path
-        const folders = workspace.workspaceFolders.map(
-            (folder) => folder.uri.fsPath,
-        );
-        folders.map(globFind);
+    private calculateEditorData(textEditor: TextEditor): {relativePath: string, workspaceFolder: string} | undefined {
+        // calculate normalize
+        const fileName = textEditor.document.fileName;
+        const editorFileUri = Uri.file(fileName);
+        const workspaceFolder = workspace.getWorkspaceFolder(editorFileUri);
+        if (!workspaceFolder) { return; } // file is not in workspace - skip it
+        const workspaceFsPath = workspaceFolder.uri.fsPath;
+        const editorFileAbs = normalizeFileName(fileName);
+        const workspaceFile = normalizeFileName(workspaceFsPath);
+        const editorFileRelative = editorFileAbs.substring(workspaceFile.length);
+        const workspaceFolderName = workspaceFolder.name;
+        return { relativePath: editorFileRelative, workspaceFolder: workspaceFolderName};
+    }
+    /**
```

Comment:
minor code styling, we just need a return between the two functions

Reply:
Can you please provide me with example - not really sure what to change here.

Reply:
https://github.com/ryanluker/vscode-coverage-gutters/pull/230/commits/9f58903e20264ff3ff7820542fc7bf4480eb4141 added it for you.

Reply:
Lost in translation  - return equals newline :)